bin/publish.sh

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env bash
+
+set -e
+
+dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
+cd "$dir"
+
+# ensure working copy is clean
+
+if [[ ! -z "$(git status --porcelain)" ]]; then
+  >&2 echo "ERROR: your git working copy is dirty"
+  exit 1
+fi
+
+NAME=$(basename "$0")
+
+function usage() {
+  local ret=${1:-0}
+  >&2 cat <<-EOF
+USAGE $ ./${NAME} -r|--repository PATH/TO/SVN/REPO -v|--version VERSION
+
+WHERE
+  repository  Path to local working copy of
+              https://plugins.svn.wordpress.org/drip/
+
+  version     Already released git tag of version to publish.
+              Use bin/release.sh to create and release new tag on github
+              prior to running this script.
+EOF
+  exit "$ret"
+}
+
+while (( "$#" )); do
+  case "$1" in
+    -r|--repository)
+      REPOSITORY=$2
+      shift 2
+      ;;
+    -v|--version)
+      VERSION=$2
+      shift 2
+      ;;
+    --) # end argument parsing
+      shift
+      break
+      ;;
+    -*) # unsupported flags
+      >&2 echo "ERROR: Unsupported flag $1"
+      usage 1
+      ;;
+    *) # preserve positional arguments
+      PARAMS="$PARAMS $1"
+      shift
+      ;;
+  esac
+done
+
+# validate repository
+
+if [[ -z "$REPOSITORY" ]]; then
+  >&2 echo "ERROR: repository is required"
+  usage 1
+elif [[ ! -d "$REPOSITORY" ]]; then
+  >&2 echo "ERROR: repository dir $REPOSITORY does not exist"
+  usage 1
+fi
+
+# validate version
+
+if [[ -z "$VERSION" ]]; then
+  >&2 echo "ERROR: version is required"
+  usage 1
+fi
+
+# update repository
+
+cd "$REPOSITORY"
+
+if ! svn status -q; then
+  >&2 echo "ERROR: your svn working copy is dirty"
+  usage 1
+fi
+
+svn update >/dev/null 2>&1
+
+cd "$dir"
+
+# check out tag locally for copy to svn repo
+
+git fetch origin --tags >/dev/null 2>&1
+current_branch=$(git rev-parse --abbrev-ref HEAD)
+tag_branch="publish_$(echo "$VERSION" | tr . _)"
+function cleanup() {
+  git checkout "$current_branch"
+  git branch -d "$tag_branch" 2>/dev/null
+
+}
+trap cleanup EXIT
+git checkout -b "$tag_branch" "tags/$VERSION" >/dev/null 2>&1
+
+# Check that tag does not already exist in svn repository
+dst="$REPOSITORY/tags/$VERSION"
+if [[ -d "$dst" ]]; then
+  >&2 echo "ERROR: tag for version in repository already exists: $dst"
+  exit 1
+fi
+
+mkdir -p "$dst"
+
+cp readme.txt "$REPOSITORY/trunk"
+cp -a \
```

Comment:
If you want to, you could use `git archive $VERSION | tar --include="file listing" -xf` to avoid fiddling with the git repo.

Reply:
Good call. This is much nicer. :1st_place_medal: